### PR TITLE
add ipo, hinge and cpo loss to dpo trainer

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?

Add some alternative pref-tuning losses to script

### CODE REVIEW

File: examples/research_projects/diffusion_dpo/train_diffusion_dpo.py [hunk 1]
```diff
@@ -299,9 +299,15 @@ def parse_args(input_args=None):
     parser.add_argument(
         "--beta_dpo",
         type=int,
-        default=5000,
+        default=2500,
```
Comment (sayakpaul) [inline examples/research_projects/diffusion_dpo/train_diffusion_dpo.py:302]: Is that reasonable?

File: examples/research_projects/diffusion_dpo/train_diffusion_dpo.py [hunk 2]
```diff
@@ -858,12 +864,19 @@ def collate_fn(examples):
                 accelerator.unwrap_model(unet).enable_adapters()
 
                 # Final loss.
-                scale_term = -0.5 * args.beta_dpo
-                inside_term = scale_term * (model_diff - ref_diff)
-                loss = -1 * F.logsigmoid(inside_term).mean()
```
Comment (sayakpaul) [inline examples/research_projects/diffusion_dpo/train_diffusion_dpo.py:863]: Can we guard this loss as well? I don't see the `scale_term` being used here. Maybe elaborate a bit?

Comment (kashif) [reply examples/research_projects/diffusion_dpo/train_diffusion_dpo.py:863]: right, so scale term no is the `beta_dpo * .5 = 2500` and the neg goes in the bracket so `(ref_diff - model_diff)` meaning the loss becomes `-1 * F.logsigmoid(args.beta_dpo * (ref_diff - model_diff))` so the orignal loss stays the same
